apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/AspNetCoreSnippets.cs

```diff
@@ -81,35 +81,33 @@ public void ReadFile(IExceptionLogger exceptionLogger)
         // End sample
 
         // Sample: RegisterGoogleLogger
-        public void Configure(ILoggerFactory loggerFactory)
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
         {
             string projectId = "[Google Cloud Platform project ID]";
-            loggerFactory.AddGoogle(projectId);
+            loggerFactory.AddGoogle(app.ApplicationServices, projectId);
         }
         // End sample
 
-#if NETCOREAPP2_0
         private class Startup { }
-        public void Configure()
+        public static void Program_Main()
         {
-            // Sample: RegisterGoogleLogger2
```

Comment:
I'd still like to keep this example and add an additional one for UseGoogleDiagnostics

Reply:
Would you like a sample where `null` for `IServiceProvider` is passed? Eg.:

```cs
var webHost = new WebHostBuilder()
    .ConfigureLogging((hostingContext, logging) =>
        logging.AddProvider(GoogleLoggerProvider.Create(null, projectId)));
```

We could change it to this though:

```cs
var webHost = new WebHostBuilder()
    .ConfigureServices(services =>
    {
        services.AddSingleton<ILoggerProvider>(sp => GoogleLoggerProvider.Create(sp, projectId));
    });
```

However that would only work in ASP.NET Core 2.0 from higher.

Reply:
Agreed we should not be showing them passing in null :)

I'm good with the second option, this sample was just showing uses the 2.0 way of doing this so this is okay with me.